app/dot/cpp/repository/services/RepositoryService.java

```diff
@@ -106,10 +113,8 @@ private boolean isCollectionInDatabase(String collectionName, MongoDatabase data
     return collectionNames.into(new ArrayList<>()).contains(collectionName);
   }
 
-  public String getSchema(
-      Class<? extends BaseEntity> entityClass, SchemaGeneratorConfig schemaGeneratorConfig) {
-    final var generator = new SchemaGenerator(schemaGeneratorConfig);
-    final var jsonSchemaAsObjectNode = generator.generateSchema(entityClass);
+  public String getSchema(Class<? extends BaseEntity> entityClass) {
+    final var jsonSchemaAsObjectNode = getSchemaGenerator().generateSchema(entityClass);
```

Comment:
The schema generator gets initialized with every call to getSchema in this way, which is not ideal, but you can save it in class field and initialize only in the constructor, or you can call getSchemaGenerator() like getSchemaGeneratorConfig() was before and pass it as a parameter.